apis/v1beta1/validation/httproute.go

```diff
@@ -287,3 +289,46 @@ func hasExactlyOnePrefixMatch(matches []gatewayv1b1.HTTPRouteMatch) bool {
 
 	return true
 }
+
+// ValidateHTTPParentRefs validates that if ParentRefs includes 2 or more references
+// to the same parent (based on kind, name, and namespace), those ParentRefs must
+// specify unique SectionName values.
+func ValidateHTTPParentRefs(parentRefs []gatewayv1b1.ParentReference, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if len(parentRefs) <= 1 {
+		return nil
+	}
+	type sameKindParentRefs struct {
+		name      gatewayv1b1.ObjectName
+		namespace gatewayv1b1.Namespace
+		kind      gatewayv1b1.Kind
+	}
+	parentRefsSectionMap := make(map[sameKindParentRefs][]gatewayv1b1.SectionName)
+	for i, p := range parentRefs {
+		namespace := new(gatewayv1b1.Namespace)
+		kind := new(gatewayv1b1.Kind)
+		section := new(gatewayv1b1.SectionName)
+		if p.Namespace != nil {
+			namespace = p.Namespace
+		}
+		if p.Kind != nil {
+			kind = p.Kind
+		}
+		if p.SectionName != nil {
+			section = p.SectionName
+		}
+		targetParentRefs := sameKindParentRefs{name: p.Name, namespace: *namespace, kind: *kind}
```

Comment:
```suggestion
		targetParentRefs := sameKindParentRefs{name: p.Name, namespace: new(gatewayv1b1.Namespace), kind: new(gatewayv1b1.Kind)}
		if p.Namespace != nil {
			targetParentRefs.namespace = *p.Namespace
		}
		if p.Kind != nil {
			targetParentRefs.kind = *p.Kind
		}
		if p.SectionName != nil {
			targetParentRefs.section = *p.SectionName
		}
```
I meant something like this, I'm not sure it compiles, but still. Does it make sense to you?

Reply:
Yes, it makes sense. I've updated like that.